homeassistant/components/teslemetry/coordinator.py

```diff
@@ -195,6 +197,17 @@ def __init__(self, hass: HomeAssistant, api: EnergySpecific) -> None:
         )
         self.api = api
 
+    async def async_config_entry_first_refresh(self) -> None:
+        """Set up the data coordinator."""
+        await super().async_config_entry_first_refresh()
+
+        # Calculate seconds until next 5 minute period plus a random delay
+        delta = randint(310, 330) - (int(time()) % 300)
+        self.logger.debug("Scheduling next %s refresh in %s seconds", self.name, delta)
+        self.update_interval = timedelta(seconds=delta)
+        self._schedule_refresh()
```

Comment:
I thought we had better functions to do this. I thought I once saw @gjohansson-ST comment about that, but not sure

Reply:
I suggested to turn off polling and setup an own caller instead.

Btw; `async_config_entry_first_refresh()` should not be overwritten. Use `_async_setup()` method instead